app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,196 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.net.Uri;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.MalformedInputException;
+
+import javax.net.ssl.HttpsURLConnection;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView widgetViewWeather;
+    private EditText widgetEditCity;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        widgetViewWeather = findViewById(R.id.msg_temperature);
+        widgetEditCity = findViewById(R.id.msg_city);
+    }
+
+    public void onClickByCity(View view) {
+        // TODO: keep this button disabled while current request not finished
+        Log.d("myTag", "OnClick start");
+
+        final URL request;
+        try {
+            String city = widgetEditCity.getText().toString();
+            checkCityString(city);
+
+            request = buildRequestUrlWithCity(city);
+            Log.d("myTag", String.valueOf(request));
+        } catch (MalformedURLException | NullPointerException | IllegalArgumentException e) {
+            e.printStackTrace();
+            msgOnWrongCity();
+            return;
+        }
+
```

Comment:
```suggestion
// todo: use Executors.newSingleThreadExecutor() executor to run network task on a separate thread.
```
let's have it as a todo for now. It will be better to implement it in the separate PR.